upup/pkg/fi/cloudup/vspheretasks/attachiso.go

```diff
@@ -131,6 +137,9 @@ func createUserData(startupStr string, dir string, dnsServer string) error {
 	dnsUpdateStr := strings.Join(lines, "\n")
 	data = strings.Replace(data, "$DNS_SCRIPT", dnsUpdateStr, -1)
 
+	vmUUIDStr := "       \"" + vmUUID + "\"\n"
```

Comment:
Nit: I am guessing you need quotes here because cloud-config needs that format. If yes, I would recommend just removing the quotes and keeping all cloud-config formatting limited to nodeup/pkg/model/cloudconfig.go